packages/webpack/src/plugin.js

```diff
@@ -429,22 +433,38 @@ class LavaMoatPlugin {
           }
         )
 
-        const HtmlWebpackPluginInUse = compiler.options.plugins.find(
-          (plugin) => plugin && plugin.constructor.name === 'HtmlWebpackPlugin'
-        )
-
-        // TODO: avoid minification
-        compilation.hooks.processAssets.tap(
-          {
-            name: PLUGIN_NAME,
-            stage: Compilation.PROCESS_ASSETS_STAGE_ADDITIONAL,
-          },
-          sesEmitHook({
-            compilation,
-            HtmlWebpackPluginInUse,
-            HtmlWebpackPluginInterop: options.HtmlWebpackPluginInterop,
-          })
-        )
+        if (options.inlineLockdown) {
+          compilation.hooks.processAssets.tap(
+            {
+              name: PLUGIN_NAME,
+              stage: Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE_INLINE,
+            },
+            sesPrefixFiles({
+              compilation,
+              inlineLockdown: options.inlineLockdown,
+            })
+          )
+        } else {
+          const HtmlWebpackPluginInUse = compiler.options.plugins.find(
+            /**
+             * @param {unknown} plugin
+             * @returns {plugin is import('webpack').WebpackPluginInstance}
+             */
+            (plugin) =>
+              !!plugin && plugin.constructor.name === 'HtmlWebpackPlugin'
```

Comment:
```suggestion
              !!plugin && typeof plugin === 'object' && plugin.constructor.name === 'HtmlWebpackPlugin'
```

Or, would `plugin instanceof HtmlWebpackPlugin` work?  Do we have the class available?

Reply:
I don't trust it to not diverge on webpack versions. but it might be unfounded.